src/overdub.jl

```diff
@@ -529,37 +529,14 @@ if VERSION >= v"1.4.0-DEV.304"
     end
 end
 
-function overdub_definition(line, file)
-    return quote
-        function $Cassette.overdub($OVERDUB_CONTEXT_NAME::$Cassette.Context, $OVERDUB_ARGUMENTS_NAME...)
-            $(Expr(:meta, :generated_only))
-            $(Expr(:meta,
-                   :generated,
-                   Expr(:new,
-                        Core.GeneratedFunctionStub,
-                        :__overdub_generator__,
-                        Any[:overdub, OVERDUB_CONTEXT_NAME, OVERDUB_ARGUMENTS_NAME],
-                        Any[],
-                        line,
-                        QuoteNode(Symbol(file)),
-                        true)))
-        end
-        function $Cassette.recurse($OVERDUB_CONTEXT_NAME::$Cassette.Context, $OVERDUB_ARGUMENTS_NAME...)
-            $(Expr(:meta, :generated_only))
-            $(Expr(:meta,
-                   :generated,
-                   Expr(:new,
-                        Core.GeneratedFunctionStub,
-                        :__overdub_generator__,
-                        Any[:recurse, OVERDUB_CONTEXT_NAME, OVERDUB_ARGUMENTS_NAME],
-                        Any[],
-                        line,
-                        QuoteNode(Symbol(file)),
-                        true)))
-        end
-    end
+
+@generated function overdub(ctx::Context, args...)
```

Comment:
Elsewhere we do:

```
@eval _overdub_fallback($OVERDUB_CONTEXT_NAME, $OVERDUB_ARGUMENTS_NAME...) = fallback($OVERDUB_CONTEXT_NAME, $OVERDUB_ARGUMENTS_NAME...)
```

I think the issue becomes if we have:
```
f(args...) = args
overdub(Ctx, f, args...)
```

this gets turned into:
```
julia> @code_lowered overdub(Ctx(), f, 1, 2)
CodeInfo(
    @ REPL[4]:1 within `f'
1 ─      #self# = Core.getfield(##overdub_arguments#423, 1)
│   %2 = Core.getfield(##overdub_arguments#423, 2)
│   %3 = Core.getfield(##overdub_arguments#423, 3)
│        args = Core.tuple(%2, %3)
└──      return args
)
```

See how the restructuring of `##overdub_arguments#423` into args happens? Even on your branch that still happens. So we working around this in the pass test seems wrong.

Reply:
Hm, I'm not sure I understand what you're saying. `_overdub_fallback` and `fallback` need to be fixed for sure. 

Are you saying we can't do this without reimplementing hygiene ourselves with `eval`?

Reply:
Since you had to fix a pass, I don't believe so.